synapse/rest/client/versions.py

```diff
@@ -44,10 +48,24 @@ def on_GET(self, request):
                     "r0.5.0",
                 ],
                 # as per MSC1497:
-                "unstable_features": {"m.lazy_load_members": True},
+                "unstable_features": {
+                    "m.lazy_load_members": True,
+                    # Advertise to clients whether they need not include an `id_server`
+                    # parameter during registration or password reset, as Synapse now decides
+                    # itself which identity server to use (or none at all).
+                    #
+                    # This is also used by a client when they wish to bind a 3PID to their
+                    # account, but not bind it to an identity server, the endpoint for which
+                    # also requires `id_server`. If the homeserver is handling 3PID
+                    # verification itself, there is no need to ask the user for `id_server` to
+                    # be supplied.
+                    "m.require_identity_server": (
```

Comment:
is this discussed in an MSC somewhere? I'm not quite sure that `unstable_features` is the right place for it.

Reply:
It came about from this discussion: https://github.com/matrix-org/matrix-doc/pull/2233#discussion_r314689461 and further discussion on Matrix.

The reason for this is we need to inform clients that this version of Synapse is new enough that it doesn't need `id_server` provided to it in various homeserver requests relating to identity.

It is a Synapse-specific thing and will be removed once enough time has passed.

Reply:
fair enough. The ideal way of representing that would be to have a new C-S version (say r0.6.0) which says that `id_server` is optional on those endpoints. Clients can then check the `versions` list.

So this is the right place for now :)